src/appUtils.ts

```diff
@@ -0,0 +1,314 @@
+import * as fs from "fs-extra";
```

Comment:
This file seems to duplicate a lot of what we've built in https://github.com/firebase/firebase-tools/blob/master/src/dataconnect/appFinder.ts - we really ought to unify the 2. I recall @fredzqm was looking at doing something like this

Reply:
I have followup PR to do just that. I had some of that in here, but Fred recommended I take it out. See the conversation here: https://github.com/firebase/firebase-tools/pull/9236#discussion_r2400008522